docs/SPEC.md

```diff
@@ -1105,6 +1128,41 @@ Examples:
     f()
     a
 
+#### Testcase statements
+
+>NOTE: Testcase statements only work within the context of a Flux developement environment. We expect to expand their use in the future.
+
+A statement that defines a test case.
+
+    TestcaseStatement = [ Attributes ] "testcase" identifier [ TestcaseExtention ] Block .
+    TestcaseExtention = "extends" string_lit
+
+Test cases are defined as a set of statements with special scoping rules.
+Each test case statement in a file is considered to be its own main package.
+In effect all statements in package scope and all statements contained within the test case statment are flattened into a single main package and executed.
+Use the `testing` package from the standard library to control the pass failure of the test case.
+
+Test extention augments an existing test case with more statements or attributes.
+A special function call `super()` must be made inside the body of a test case extention, in its place all statements from the parent test case will be executed.
+
+
+Examples:
+
+A basic test case for addition
+
+    import "testing"
+
+    testcase addition {
+        testing.assertEqualValues(got: 1 + 1, want: 2)
+    }
+
+An example of test case extention to validate a feature does not regress existing behavior.
+
+    @feature({vectorization: true})
```

Comment:
Should we make features of Flux explicitly boolean? Meaning you can't have a feature accept an arbitrary value like our system feature flags do? 

This would change to be
```
@feature("vectorization")
```

Naming the feature is then sufficient to know its should be set to true. 

Thoughts?

Reply:
Features can be arbitrary types so I think it's better as is.

Reply:
I think I agree.

Furthermore, for vectorization the need was often to test with multiple related flags together. If we opted for `@feature("name")`, I think we should be able to stack these attributes:

```
@feature("vectorizedMap")
@feature("vectorizedLogicalOps")
@feature("vectorizedConditionals")
testcase something_with_vec_cond_and_logical {
  // ..
}
```

I like this less than being able to specify a map of features I want to enable.